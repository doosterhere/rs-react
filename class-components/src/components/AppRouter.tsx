import { createBrowserRouter, RouterProvider } from 'react-router-dom';

import { ROUTES } from '../constants';

import { HomePage, Page404 } from '../pages';
import { ListItemDetailed } from './ListItemDetailed/ListItemDetailed';

const router = createBrowserRouter([
  {
    path: ROUTES.HOME,
    element: <HomePage />,
    errorElement: <Page404 />,
    children: [
      {
        path: ROUTES.DETAIL,
        element: <ListItemDetailed />,
      },
    ],
  },
]);

const AppRouter = () => <RouterProvider router={router} />;

export { AppRouter };
