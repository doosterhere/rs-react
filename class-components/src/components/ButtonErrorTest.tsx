import { useState } from 'react';

const ButtonErrorTest = () => {
  const [hasError, setHasError] = useState(false);

  const handleThrowError = () => {
    setHasError(true);
  };

  if (hasError) {
    throw new Error('Throwing the error while rendering');
  }

  return (
    <>
      <button type="button" className="button-error" onClick={handleThrowError}>
        Throw an error
      </button>
    </>
  );
};

export default ButtonErrorTest;
