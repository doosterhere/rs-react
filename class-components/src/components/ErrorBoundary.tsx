import { Component, ReactNode } from 'react';

interface IErrorBoundaryProps {
  children: ReactNode;
}

class ErrorBoundary extends Component<IErrorBoundaryProps> {
  state = {
    hasError: false,
    errorMessage: '',
  };

  static getDerivedStateFromError(error: Error) {
    return { hasError: true, errorMessage: error.message.toString() };
  }

  componentDidCatch(error: Error, errorInfo: React.ErrorInfo) {
    console.error(`Error in error boundary: ${error}`, `Component stack: ${errorInfo.componentStack}`);
  }

  reloadPage() {
    window.location.assign('/');
  }

  render() {
    return this.state.hasError ? (
      <div className="container" style={{ display: 'flex', alignItems: 'center' }}>
        <h1 data-testid="error-boundary">Something went wrong!</h1>
        <button onClick={this.reloadPage} style={{ width: '300px' }}>
          Return to main page
        </button>
      </div>
    ) : (
      this.props.children
    );
  }
}

export { ErrorBoundary };
