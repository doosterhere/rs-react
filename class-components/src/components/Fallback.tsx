const Fallback = () => {
  const reloadPage = () => {
    history.go(0);
  };

  return (
    <div>
      <h1>Something went wrong!</h1>
      <button onClick={reloadPage}>Return to main page</button>
    </div>
  );
};

export default Fallback;
