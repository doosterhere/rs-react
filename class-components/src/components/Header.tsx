import { Link } from 'react-router-dom';

const Header = () => (
  <header role="heading">
    <div>
      <Link to={'/'}>
        <img src="images/star-wars-logo.png" alt="Star Wars logo" />
      </Link>
      Star Wars planet finder app
    </div>
  </header>
);

export { Header };
