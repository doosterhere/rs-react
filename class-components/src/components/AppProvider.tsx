'use client';

import { ReactNode, useRef } from 'react';
import { Provider } from 'react-redux';

import '../styles/global.css';

import { AppStore, configStore } from '../store/store';
import { ThemeProvider } from '../components/ThemeContext';
import { ErrorBoundary } from './ErrorBoundary';

function AppProvider({ children }: { children: ReactNode }) {
  const storeRef = useRef<AppStore>();

  if (!storeRef.current) {
    storeRef.current = configStore();
  }

  return (
    <Provider store={storeRef.current}>
      <ErrorBoundary>
        <ThemeProvider>{children}</ThemeProvider>
      </ErrorBoundary>
    </Provider>
  );
}

export { AppProvider };
