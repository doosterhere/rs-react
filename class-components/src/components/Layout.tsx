import { Outlet } from 'react-router-dom';
import ErrorBoundary from './ErrorBoundary';
import Fallback from './Fallback';
import SearchBar from './SearchBar';

const Layout = () => {
  return (
    <>
      <ErrorBoundary fallbackComponent={<Fallback />}>
        <header>
          <div>
            <img
              src="../../public/images/star-wars-logo.png"
              alt="Star Wars logo"
            />
            Star Wars planet finder app
          </div>
          <SearchBar />
        </header>
        <Outlet />
        <footer>2024</footer>
      </ErrorBoundary>
    </>
  );
};

export default Layout;
