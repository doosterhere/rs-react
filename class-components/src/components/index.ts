import SearchBar from './SearchBar.tsx';
import List from './List.tsx';
import ListItem from './ListItem.tsx';
import Loader from './Loader.tsx';
import ErrorBoundary from './ErrorBoundary.tsx';
import ButtonErrorTest from './ButtonErrorTest.tsx';
import Fallback from './Fallback.tsx';
import Modal from './Modal.tsx';
import AppRouter from './AppRouter.tsx';
import Layout from './Layout.tsx';

export {
  SearchBar,
  List,
  ListItem,
  Loader,
  ErrorBoundary,
  ButtonErrorTest,
  Fallback,
  Modal,
  AppRouter,
  Layout,
};
