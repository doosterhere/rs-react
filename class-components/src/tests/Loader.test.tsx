import { render, screen } from '@testing-library/react';

import { Loader } from '../components';

describe('Loader', () => {
  it('should render correctly', () => {
    const { container } = render(<Loader />);

    expect(container).toMatchSnapshot();

    expect(screen.getByRole('progressbar')).toBeInTheDocument();
  });
});
