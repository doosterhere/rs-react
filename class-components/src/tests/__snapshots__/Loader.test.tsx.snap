// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render correctly 1`] = `
<div>
  <div
    class="overlay"
    role="progressbar"
  >
    <div
      class="loader"
    />
  </div>
</div>
`;
